crates/commons/src/message.rs

```diff
@@ -43,10 +44,38 @@ pub enum Message {
     },
     TradeError {
         order_id: Uuid,
-        error: String,
+        error: TradingError,
     },
 }
 
+#[derive(Serialize, Deserialize, Clone, Error, Debug, PartialEq)]
+pub enum TradingError {
+    #[error("Invalid order: {0}")]
+    InvalidOrder(String),
+    #[error("{0}")]
```

Comment:
:wrench: I think this should be

```suggestion
    #[error("No match found: {0}")]
```